Toggle errata

boutell opened the pull request:
<!-- Please don't delete this template -->

<!-- PULL REQUEST TEMPLATE -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

## Summary

PR to your PR to my PR:

* If an item has errors always toggle it open
* If there is no title field open all the items and do not offer a toggle (probably most natural for single field scenarios)

## What are the specific steps to test this change?

* Make a subfield required and mess about with saving the document without completing it
* Check out the behavior with a single subfield and no titleField configured, see if you like my tweak there

## What kind of change does this PR introduce?
*(Check at least one)*

- [X] Bug fix
- [X] New feature
- [ ] Refactor
- [ ] Documentation
- [ ] Build-related changes
- [ ] Other

## Make sure the PR fulfills these requirements:

- [X] It includes a) the existing issue ID being resolved, b) a convincing reason for adding this feature, or c) a clear description of the bug it resolves
- [ ] The changelog is updated
- [ ] Related documentation has been updated
- [ ] Related tests have been updated

If adding a new feature without an already open issue, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

stuartromanek commented:
Not sure I agree the absence of `titleField` connotes 'expand all items all the time' .. to me it would make sense to
- keep schemas expanded when the array item schema is under a certain length (2 or less items?)
- an explicit flag in the schema def, like
 ```
footerLinks: {
  label: 'Footer Links',
  type: 'array',
  inline: {
    expand: false
  },
  fields: { ... }
},
 ```
The flag feels a little extra since we're hustling on this but idk, thoughts?

stuartromanek commented:
To clarify ^ i think those two bullets would be used together.. 
If an inline array is present with under 3 child fields, auto expand the schemas unless the flag is included and says otherwise.
If an inline array is present with greater than 3 child fields, auto collapse the schemas unless the flag is included and says otherwise.

boutell commented:
* Implemented default behavior per Stu's suggestion
* Implemented the sub-option to override as suggested too, but called it `alwaysExpand` because I kept getting confused about whether `expand: true` meant "expandable" or "always expanded," this is unambiguous
